src/composeappmanager.cc

```diff
@@ -456,6 +462,34 @@ void ComposeAppManager::handleRemovedApps(const Uptane::Target& target) const {
   }
 }
 
+void ComposeAppManager::stopDisabledComposeApps(const Uptane::Target& target) const {
+  if (!boost::filesystem::is_directory(cfg_.apps_root)) {
+    LOG_DEBUG << "cfg_.apps_root does not exist";
+    return;
+  }
+
+  // an intersection of apps specified in Target and the configuration
+  // i.e. the apps that are supposed to be installed and running
+  const auto& current_apps = getApps(target);
+
+  for (auto& entry : boost::make_iterator_range(boost::filesystem::directory_iterator(cfg_.apps_root), {})) {
+    if (boost::filesystem::is_directory(entry)) {
+      std::string name = entry.path().filename().native();
+      if (current_apps.find(name) == current_apps.end()) {
+        LOG_WARNING << "Docker Compose App(" << name
+                    << ") installed, "
+                       "but is either removed from configuration or not defined in current Target. "
+                       "Stopping it";
+
+        // I have no idea via the package manager interface method install() is const which is not a const
+        // method by its definition/nature
+        auto& non_const_app_engine = (const_cast<ComposeAppManager*>(this))->app_engine_;
+        non_const_app_engine->stop({name, ""});
+      }
+    }
+  }
+}
+
 // Handle the case like:
 //  1) sota.toml is configured with 2 compose apps: "app1, app2"
 //  2) update is applied, so we are now running both app1 and app2
```

Comment:
we now have `removeDisabledComposeApps` and `stopDisabledComposeApps`. They almost do the same thing. Do we need both?

Reply:
Yes, we need both.

Reply:
I brought it to a common method.